robot-name/Pratster95/robot_name.go

```diff
@@ -0,0 +1,40 @@
+package robotname
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+// Define the Robot type here.
+type Robot struct {
+	name string
+}
+
+var nameUsed = map[string]bool{}
+
+const max = 26 * 26 * 10 * 10 * 10
+
+func (r *Robot) Name() (string, error) {
+	if r.name != "" {
+		return r.name, nil
+	}
+	if len(nameUsed) >= max {
+		return "", fmt.Errorf("No Names left")
+	}
+	for {
+		name := getName()
+		if !nameUsed[name] {
+			nameUsed[name] = true
+			r.name = name
+			return r.name, nil
+		}
+	}
+}
+
+func (r *Robot) Reset() {
+	r.name = ""
+}
+func getName() string {
+	return fmt.Sprintf("%v%v%v%v%v", string(rand.Int31n(26)+66), string(rand.Int31n(26)+66), rand.Int31n(10), rand.Int31n(10), rand.Int31n(10))
```

Comment:
I'd say that `%v` is too generic and that is better to use `%s%s%d%d%d` in this case. Another change for this line would be to generate a single random number for the numerical part. So something like this:
```suggestion
	return fmt.Sprintf("%s%s%d", string(rand.Intn(26)+66), string(rand.Intn(26)+66), rand.Intn(1000))
```